app/Http/Controllers/ContestsController.php

```diff
@@ -37,9 +37,13 @@ public function show($id)
             abort(404);
         }
 
-        return view("contests.{$contest->type}")
-                ->with('contest', $contest)
-                ->with('contestJson', $contest->defaultJson($user));
+        if ($contest->isShowingEntries()) {
```

Comment:
got a suggestion for a better name? it's a hard one since it encapsulates voting, post-voting and results